### PR TITLE
Rebound: add rev2 and thus rev1 as well

### PR DESCRIPTION
**Pull request by Rossman360**
I made a rev2 for Rebound, a 49-key version. Then, I made rev1 files for the 48-key version and created default keymaps for each

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/montsinger/rebound/keymaps/default49/keymap.c [hunk 1]
```diff
@@ -0,0 +1,163 @@
+#include QMK_KEYBOARD_H
+
+// Each layer gets a name for readability, which is then used in the keymap matrix below.
+// The underscores don't mean anything - you can have a layer called STUFF or any other name.
+// Layer names don't all need to be of the same length, obviously, and you can also skip them
+// entirely and just use numbers.
+
+enum layer_names
+{
```
Comment (drashna) [inline keyboards/montsinger/rebound/keymaps/default49/keymap.c:9]: Just a nit pick: 
```suggestion
enum layer_names {
```

File: keyboards/montsinger/rebound/rev2/rules.mk [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+
+LAYOUTS = ortho_5x12
+LAYOUTS_HAS_RGB=no
```
Comment (drashna) [inline keyboards/montsinger/rebound/rev2/rules.mk:32]:
```suggestion
LAYOUTS_HAS_RGB = no
```

**Comment by drashna** [comment]
Okay, taking a closer look at this, both revisions lack a "default" keymap. 

Additionally, the revision 2 errors out with the default community layout (too many keys).   And it looks like it doesn't actually match the 5x12 or 4x12 ortho layouts.    And if that is the case, support (and naming) for it should be removed.

**Comment by Rossman360** [comment]
"buncha stuff" PR

-renamed ortho_5x12 to "49" and ortho_4x12 to "48" to reflect the number of keys rather than how the matrix is set up.
-made rev2 default
-added rotary pins for rev2
-renamed default49 keymap to default
-set up rev1 so it doesn't fail with 49-key keymap (still unsure how to set a default keymap for a specific rev)

File: keyboards/montsinger/rebound/rev2/rev2.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ * The first section contains all of the arguments representing the 
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+ 
+#define LAYOUT_ortho_5x12( \
```
Comment (zvecr) [inline keyboards/montsinger/rebound/rev2/rev2.h:29]: This is not `ortho_5x12 compatible
```suggestion
#define LAYOUT( \
```

File: keyboards/montsinger/rebound/rev2/rules.mk [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+
+LAYOUTS = ortho_5x12
+LAYOUTS_HAS_RGB = no
```
Comment (zvecr) [inline keyboards/montsinger/rebound/rev2/rules.mk:32]:
```suggestion
```

File: keyboards/montsinger/rebound/rules.mk [hunk 5]
```diff
@@ -28,5 +28,7 @@ AUDIO_ENABLE = no           # Audio output on port C6
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
 HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
 
-LAYOUTS = ortho_4x12
-LAYOUTS_HAS_RGB=no
+LAYOUTS = ortho_5x12
+LAYOUTS_HAS_RGB = no
+
```
Comment (zvecr) [inline keyboards/montsinger/rebound/rules.mk:33]:
```suggestion
```

File: keyboards/montsinger/rebound/rev1/rules.mk [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
+
+LAYOUTS = 48
+LAYOUTS_HAS_RGB=no
```
Comment (zvecr) [inline keyboards/montsinger/rebound/rev1/rules.mk:32]:
```suggestion

LAYOUTS = ortho_4x12
```

File: keyboards/montsinger/rebound/rev1/rev1.h [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+/* This is a shortcut to help you visually see your layout.
+ * The first section contains all of the arguments representing the 
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+ 
+#define LAYOUT_48( \
```
Comment (zvecr) [inline keyboards/montsinger/rebound/rev1/rev1.h:29]:
```suggestion
#define LAYOUT_ortho_4x12 \
```

**Comment by Rossman360** [comment]
hang on, I'm changing everything

File: keyboards/montsinger/rebound/rebound.h [hunk 8]
```diff
@@ -37,3 +43,16 @@
   { K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B }, \
   { K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B } \
 }
+
+#define LAYOUT_49( \
```
Comment (zvecr) [inline keyboards/montsinger/rebound/rebound.h:47]: Is this compatible with both rev1 and rev2?

File: keyboards/montsinger/rebound/rebound.h [hunk 9]
```diff
@@ -26,7 +32,7 @@
  * represents the switch matrix.
  */
  
-#define LAYOUT_ortho_4x12( \
+#define LAYOUT_48( \
```
Comment (zvecr) [inline keyboards/montsinger/rebound/rebound.h:35]: Is this compatible with both rev1 and rev2?

Also, this should still be ortho_4x12 based
```suggestion
#define LAYOUT_ortho_4x12( \
```

File: keyboards/montsinger/rebound/rebound.h [hunk 10]
```diff
@@ -14,6 +14,12 @@
...
 
+#ifdef KEYBOARD_montsinger_rebound_rev1
+    #include "rev1.h"
+#elif KEYBOARD_montsinger_rebound_rev2
+    #include "rev2.h"
+#endif
+
 #pragma once 
 
 #include "quantum.h"
```
Comment (zvecr) [inline keyboards/montsinger/rebound/rebound.h:25]:
```suggestion
#pragma once 
#include "quantum.h"

#ifdef KEYBOARD_montsinger_rebound_rev1
    #include "rev1.h"
#elif KEYBOARD_montsinger_rebound_rev2
    #include "rev2.h"
#8397 
```

File: keyboards/montsinger/rebound/keymaps/rossman360/keymap.c [hunk 11]
```diff
@@ -16,40 +16,42 @@ enum layer_names {
...
+
+
 const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
 
-[_BASE] = LAYOUT_ortho_4x12(
-    KC_ESC  , KC_Q   , KC_W   , KC_E   , KC_R   , KC_T   ,   KC_Y   , KC_U   , KC_I   , KC_O   , KC_P   , NTAB  ,
-    JUMPBACK, KC_A   , KC_S   , KC_D   , KC_F   , KC_G   ,   KC_H   , KC_J   , KC_K   , KC_L   , KC_SCLN, KC_QUOT,
-    KC_LSFT , KC_Z   , KC_X   , KC_C   , KC_V   , KC_B   ,   KC_N   , KC_M   , KC_COMM, KC_DOT , KC_SLSH, KC_MINS,
-    CMDBSP  , ALTDEL , KC_NO  , CTRLSP , TABMOD , PGMOD  ,   ENTMOD , SPCMOD , MO(_DEL),KC_NO  , KC_BSPC, KC_DEL
+[_BASE] = LAYOUT_ortho_5x12(
```
Comment (zvecr) [inline keyboards/montsinger/rebound/keymaps/rossman360/keymap.c:23]: None of these add an additional row to match the layout macro

**Comment by Rossman360** [comment]
I redid everything so that there is only one rev, added an info.json with both available layouts, renamed the layouts to LAYOUT (49 key) and LAYOUT_48. The default48 keymap disables the rotary encoder and utilizes the appropriate layout

File: keyboards/montsinger/rebound/rules.mk [hunk 12]
```diff
@@ -28,5 +28,5 @@ AUDIO_ENABLE = no           # Audio output on port C6
 FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
 HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
 
-LAYOUTS = ortho_4x12
-LAYOUTS_HAS_RGB=no
+LAYOUTS_HAS_RGB = no
```
Comment (zvecr) [inline keyboards/montsinger/rebound/rules.mk:31]:
```suggestion
LAYOUTS = ortho_4x12
LAYOUTS_HAS_RGB = no
```

File: keyboards/montsinger/rebound/info.json [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+		{"x":6, "y":4},
+                {"x":7, "y":4},
+                {"x":8, "y":4},
+                {"x":9, "y":4},
+                {"x":10, "y":4},
+                {"x":11, "y":4},
+                {"x":12, "y":4}
+            ]
+        },
+        "LAYOUT_48": {
```
Comment (zvecr) [inline keyboards/montsinger/rebound/info.json:67]:
```suggestion
        "LAYOUT_ortho_4x12": {
```

File: keyboards/montsinger/rebound/keymaps/default48/keymap.c [hunk 14]
```diff
@@ -0,0 +1,163 @@
...
+ * | Tab  |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Bksp |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Esc  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |Adjust| Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_48(
```
Comment (zvecr) [inline keyboards/montsinger/rebound/keymaps/default48/keymap.c:40]:
```suggestion
[_QWERTY] = LAYOUT_ortho_4x12(
```

File: keyboards/montsinger/rebound/keymaps/default48/keymap.c [hunk 15]
```diff
@@ -0,0 +1,163 @@
...
+ * | Tab  |   Q  |   W  |   F  |   P  |   G  |   J  |   L  |   U  |   Y  |   ;  | Bksp |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Esc  |   A  |   R  |   S  |   T  |   D  |   H  |   N  |   E  |   I  |   O  |  '   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   Z  |   X  |   C  |   V  |   B  |   K  |   M  |   ,  |   .  |   /  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |Adjust| Ctrl | Alt  | GUI  |Lower |Space |Space |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_COLEMAK] = LAYOUT_48(
```
Comment (zvecr) [inline keyboards/montsinger/rebound/keymaps/default48/keymap.c:58]:
```suggestion
[_COLEMAK] = LAYOUT_ortho_4x12(
```

File: keyboards/montsinger/rebound/keymaps/default48/keymap.c [hunk 16]
```diff
@@ -0,0 +1,163 @@
...
+ * | Tab  |   '  |   ,  |   .  |   P  |   Y  |   F  |   G  |   C  |   R  |   L  | Bksp |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Esc  |   A  |   O  |   E  |   U  |   I  |   D  |   H  |   T  |   N  |   S  |  /   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   ;  |   Q  |   J  |   K  |   X  |   B  |   M  |   W  |   V  |   Z  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |Adjust| Ctrl | Alt  | GUI  |Lower |Space |Space |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_DVORAK] = LAYOUT_48(
```
Comment (zvecr) [inline keyboards/montsinger/rebound/keymaps/default48/keymap.c:76]:
```suggestion
[_DVORAK] = LAYOUT_ortho_4x12(
```

File: keyboards/montsinger/rebound/keymaps/default48/keymap.c [hunk 17]
```diff
@@ -0,0 +1,163 @@
...
+ * |   ~  |   !  |   @  |   #  |   $  |   %  |   ^  |   &  |   *  |   (  |   )  | Del  |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Del  |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |   _  |   +  |   {  |   }  |  |   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |ISO ~ |ISO | |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT_48(
```
Comment (zvecr) [inline keyboards/montsinger/rebound/keymaps/default48/keymap.c:94]:
```suggestion
[_LOWER] = LAYOUT_ortho_4x12(
```

File: keyboards/montsinger/rebound/keymaps/default48/keymap.c [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+ * |   `  |   1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  | Del  |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Del  |  F1  |  F2  |  F3  |  F4  |  F5  |  F6  |   -  |   =  |   [  |   ]  |  \   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |  F7  |  F8  |  F9  |  F10 |  F11 |  F12 |ISO # |ISO / |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      | Next | Vol- | Vol+ | Play |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_RAISE] = LAYOUT_48(
```
Comment (zvecr) [inline keyboards/montsinger/rebound/keymaps/default48/keymap.c:112]:
```suggestion
[_RAISE] = LAYOUT_ortho_4x12(
```

File: keyboards/montsinger/rebound/keymaps/default48/keymap.c [hunk 19]
```diff
@@ -0,0 +1,163 @@
...
+ * |      | Reset|      |      |      |      |      |      |      |      |      |  Del |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * |      |      |      |Aud on|Audoff|AGnorm|AGswap|Qwerty|Colemk|Dvorak|      |      |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * |      |      |      |      |      |      |      |      |      |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] =  LAYOUT_48(
```
Comment (zvecr) [inline keyboards/montsinger/rebound/keymaps/default48/keymap.c:130]:
```suggestion
[_ADJUST] =  LAYOUT_ortho_4x12(
```

**Comment by Rossman360** [comment]
I think I've made the suggested changes, but I get errors compiling the default48 keymap because of the implicit declaration of the community layout, when it wasn't doing that when I had named given it a custom name

File: keyboards/montsinger/rebound/rebound.h [hunk 20]
```diff
@@ -26,7 +26,7 @@
  * represents the switch matrix.
  */
  
-#define LAYOUT_ortho_4x12( \
+#define LAYOUT_ortho4x12( \
```
Comment (zvecr) [inline keyboards/montsinger/rebound/rebound.h:29]:
```suggestion
#define LAYOUT_ortho_4x12( \
```

Comment (Rossman360) [reply keyboards/montsinger/rebound/rebound.h]: Ah, yes, my friend Underscore, glad to see y'all've met. Thanks for the help bud!

File: keyboards/montsinger/rebound/readme.md [hunk 21]
```diff
@@ -5,11 +5,13 @@
 A conjoined-split 40% 4x12 ortholinear keyboard made and sold by Montsinger. [More info on Montsinger.net](https://montsinger.net).
 
 * Keyboard Maintainer: [Rossman360](https://github.com/rossman360)
-* Hardware Supported: Pro Micro, Elite-C
+* Hardware Supported: Rebound rev1 or rev2; Pro Micro or Elite-C
 * Hardware Availability: [Montsinger.net](https://montsinger.net)
 
 Make example for this keyboard (after setting up your build environment):
 
     make montsinger/rebound:default
```
Comment (noroadsleft) [inline keyboards/montsinger/rebound/readme.md:13]: As currently implemented, the make command must specify either `/rev1` or `/rev2`; this syntax won't work.

```suggestion
```

Comment (Rossman360) [reply keyboards/montsinger/rebound/readme.md]: the change for line 8 is how my file already looked

File: keyboards/montsinger/rebound/rev2/keymaps/default/keymap.c [hunk 22]
```diff
@@ -0,0 +1,162 @@
...
+      }
+      return false;
+    case DVORAK:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_DVORAK);
+      }
+      return false;
+  }
+  return true;
+}
```
Comment (drashna) [inline keyboards/montsinger/rebound/rev2/keymaps/default/keymap.c:162]: Just a couple of nit pick, but spacing:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
}

bool process_record_user(uint16_t keycode, keyrecord_t *record) {
    switch (keycode) {
        case QWERTY:
            if (record->event.pressed) {
                set_single_persistent_default_layer(_QWERTY);
             }
             break;
        case COLEMAK:
            if (record->event.pressed) {
                set_single_persistent_default_layer(_COLEMAK);
            }
            break;
        case DVORAK:
            if (record->event.pressed) {
                set_single_persistent_default_layer(_DVORAK);
            }
            break;
    }
    return true;
}
```

**Comment by Rossman360** [comment]
anything needed from me at this time?

**Comment by Erovia** [comment]
You could also create a keyboard level `rules.mk` and use the `DEFAULT_FOLDER` directive. With this the `rebound` build target will not brake for users.
You can check the lets_split files for more detail.

File: keyboards/montsinger/rebound/rules.mk [hunk 23]
```diff
@@ -1,32 +1 @@
...
-RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
-MIDI_ENABLE = no            # MIDI support
-BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
-AUDIO_ENABLE = no           # Audio output on port C6
-FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
-HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
-
-LAYOUTS = ortho_4x12
-LAYOUTS_HAS_RGB=no
+DEFAULT_FOLDER = montsinger/rebound/rev2
```
Comment (noroadsleft) [inline keyboards/montsinger/rebound/rules.mk:1]: IMO this should point at rev1, which maintains behaviour for the existing users.

```suggestion
DEFAULT_FOLDER = montsinger/rebound/rev1
```

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**